### PR TITLE
Viewport size change

### PR DESCRIPTION
**Pull request by jamesmuerle**

### CODE REVIEW

File: game/static/game/js/game.js [hunk 1]
```diff
@@ -221,15 +223,20 @@ Game.prototype.renderMethod = function() {
             d3 = this.gs_renderer.project(d3);
             d4 = this.gs_renderer.project(d4);
 
-            // this.ui_renderer.renderText("d1 x: " + d1.x + ", y: " + d1.y, 400, 200, "red");
-            // this.ui_renderer.renderText("d2 x: " + d2.x + ", y: " + d2.y, 400, 220, "red");
-            // this.ui_renderer.renderText("d3 x: " + d3.x + ", y: " + d3.y, 400, 240, "red");
-            // this.ui_renderer.renderText("d4 x: " + d4.x + ", y: " + d4.y, 400, 260, "red");
+            // Normalize the coordinates by the size of the map.
+            // d1.x = d1.x / map_size.width;
```
Comment (hellochar) [inline game/static/game/js/game.js]: Do we need these comments? Let's just delete them

File: lib/static/lib/js/ui_renderer.js [hunk 2]
```diff
@@ -34,6 +34,8 @@ function UIRenderer(canvas) {
     this.selection = []; //An array of currently selected units
     this.scaleRatio = 800 / 10000;
 
+    // The minimap is a square of side lengths equal to this value.
+    this.minimapSize = 200;
```
Comment (hellochar) [inline lib/static/lib/js/ui_renderer.js:51]: Is the canvas' dimensions supposed to be equal to this?

File: lib/static/lib/js/ui_renderer.js [hunk 3]
```diff
@@ -238,30 +240,99 @@ UIRenderer.prototype.renderWinOrLoseText = function(player) {
...
     this.ctx.strokeStyle = "#FFFFFF";
     this.ctx.lineWidth = 2;
     this.ctx.globalAlpha = 1;
     this.ctx.beginPath();
 
-    d1 = this.translatePos(d1.x, d1.y);
-    d2 = this.translatePos(d2.x, d2.y);
-    d3 = this.translatePos(d3.x, d3.y);
-    d4 = this.translatePos(d4.x, d4.y);
+    this.three2minimap(d1, terrainSize);
```
Comment (hellochar) [inline lib/static/lib/js/ui_renderer.js:368]: This behavior doesn't match the comments; it looks like you should be passing the world coordinates for the viewport instead?

**Comment by hellochar** [comment]
Lg2m, looking forward to integrating things together so the code doesn't have to be so hacky

File: lib/static/lib/js/ui_renderer.js (hunk 2 above)
Comment (jamesmuerle) [inline lib/static/lib/js/ui_renderer.js:51]: No. The canvas' size is irrelevant. The minimap will be a minimapSize by minimapSize square in the upper left corner of the canvas.

**Comment by jamesmuerle** [comment]
I agree. This felt really bad. I basically copied and pasted code we already had to try to get it to work.

**Comment by hellochar** [comment]
lg2m; lets merge this in after conflicts resolved

**Pull request merged**